webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/LdapQueryProcessor.scala

```diff
@@ -440,13 +440,16 @@ class InternalLDAPQueryProcessor(
    * A raw execution without log, special optimisation case, invert, etc
    */
   def rawInternalQueryProcessor(query: Query, debugId: Long = 0L): IOResult[Seq[NodeId]] = {
-    for {
-      nq  <- normalize(query).toIO.chainError("Error when normalizing LDAP query")
-      ids <- getMapDn(nq, debugId).flatMap {
-               case None      => Seq().succeed
-               case Some(dns) => executeLdapQueries(dns, nq, Seq("1.1"), debugId)
-             }
-    } yield ids
+    if (query.criteria.isEmpty) Seq().succeed
```

Comment:
That method seems to be called only in the `NodeFactQueryProcessor` so we could not have guarded for empty (since it's also added their) ; but the old semantic seems strange: if there is no criteria line, then we shouldn't get any nodes, not all nodes. 

I'm not changing it elsewhere for fear of unsuspected side effects on other implementation relying on that semantic.